### DISCUSSION TITLE
Reduce the app size

### DISCUSSION THREAD

**Discussion by Sadwhy3** [PaulWoitaschek/Voice#2042] [Ideas]
Why would a simple audiobook player app need 74~mb of space? That's highly overkill.

**Comment by PaulWoitaschek** [comment]
Because of ffmpeg. If you download it from Google play the download is only about 12mb

**Comment by shirish87** [comment]
I could be wrong but ffmpeg/ffprobe seems to be in use only to extract metadata. If that's the case, I have a little [media3-extractor-m4b](https://github.com/dotslashlabs/media3-extractor-m4b) library here that may help. I'm trying to find time to work on my own audiobook app, but hopefully this helps you!

**Reply by PaulWoitaschek** [reply]
Unfortunately that doesn't handle all the different chapter formats and weirdness especially out there in media formats

**Reply by shirish87** [reply]
I've added chapter parsing for as many `m4b` files I could get my hands on :sweat:. But yes, there are probably many more cases and formats to handle.

**Reply by PaulWoitaschek** [reply]
M4b is pretty straight forward but there are also  id3 chapters, flac chapters, mkv chapters etc etc.

**Reply by PaulWoitaschek** [reply]
Ie here are some formats:
https://github.com/PaulWoitaschek/chapterreader

But the whole topic was only solved once switching to ffmpeg

**Reply by shirish87** [reply]
Thanks! those sample files are helpful!
